### DISCUSSION TITLE
dxf export in cutting diagram | make x/y axis switchable

### DISCUSSION THREAD

**Discussion by holle75** [lairdubois/lairdubois-opencutlist-sketchup-extension#675] [Ideas]
Hi, due to sheet material is given in "length x width" in OPC, but CNC programs are usually working x/y the opposite way ... being able to switch x with y in dxf export in the cutting diagram would be nice (portrait instead of landscape). Now one have to rotate for example a standard 250 x 125cm sheet 90 degrees in the CAM Program for milling when working with a standard zeropoint (front/left). Not a biggie, but ...

I´m new to CNC milling and might haven´t got a clue what i am talking about (meaning my settings are wrong).  Please correct me ;)

**Comment by bbeaulant** [comment]
Hello,

As the DXF should be converted to G-Code or something like that in a CAM software, I think that applying a rotation in tat software is not a big action.
Adding multiple options to OpenCutList is not necessary the best option to keep it simple :)

**Comment by holle75** [comment]
Hi Boris, well, yes and no ;)

I tried Estlcam and couldn´t find an option to nicely rotate your DXF objects without running into problems with the (only visual as an aid) workplane you can define by yourself (in our case a standard multiplex sheet to find perfect Point Zero). So i have to do an interim step with a DXF Software to rotate the DXF (whole sheet).

And i have no clue how other CAM Software is working. But i could imagine, that this portrait/landscape problem exists in some of them.

BUT, you are right in your approach to keep OCL simple. Nevertheless a checkbox "swap x/y" or a dropbox "portrait/landscape" in DXF export wouldn´t make OCL sooo much more complex ;)

I would be interested if this is a problem for others too, or just for me because of using Estlcam and being a beginner in CNC´s. If it´s only me, i need to learn a better workaround, if others go crazy too i would appreciate anyone who comments here.

Thank you for your answer.

**Reply by bbeaulant** [reply]
Can you put a picture of what do you exactly mean by swapping x/y ? Where is the origin ?

_Because this will create a mirror and if it's not a problem with rectangle parts it will be with irregular parts._

> DXF export wouldn´t make OCL sooo much more complex ;)

No, but it is with this type of option that future features are more difficult to develop.

**Comment by holle75** [comment]
In our case when working with standard sheet sizes (2500x1250) in OCL x is 2500 and y is 1250mm. Thats, so to say, landscape view when exporting this sheet as DXF. 

On our CNC x/y is swapped. x is 1250 and y is 2500. I think (but don´t know) thats the same on every CNC. Thats portrait view. Front/left is Point zero. See (horrible, i´m bad with writing on a touchpad) attached image.

The option i´m looking for is "rotating the DXF output 90 degrees" (that shouldn´t be a problem in any form of output?). Yes, you can easily rotate the DXF in the CAM Software. In Estlcam you are getting a problem with finding zero when doing so and using a visual aid that shows your sheet dimensions (don´t have the CNC feeding computer at home right now to show a screenshot).

![x-y](https://github.com/user-attachments/assets/86215213-311c-480e-bed3-3449df80a4d7)

**Reply by bbeaulant** [reply]
So if origin is on the left and positive x on the right, positive y on top it's a mirror ...
What a strange and non standard configuration ...

**Comment by holle75** [comment]
Yes, might be. I only know "our CNC" (and not very well at this point in time) ;)

**Reply by bbeaulant** [reply]
Here is what Aspire Vectric says in it [doc](https://docs.vectric.com/docs/V12.0/Aspire/FRA/Help/page/user-guide/) : 

> As you look at your CNC machine, the normal conventional is that left and right movement is controlled by the X-axis, forward and backward movement controlled by the Y-axis and up and down movement is controlled by the Z-axis.
> In our software the width of your job will typically be equivalent to the X-axis of your CNC machine and the height of your job to its Y-axis.

**Comment by holle75** [comment]
Thank you for your research. I must say, that makes sense. Let me play around with the CNC and i´ll get back to this topic. At the moment there is a high possibility that i mess it up.

I didn´t it seems. Just found that topic that explains "my problem" -> https://forum.v1e.com/t/swapping-x-and-y-axis-on-a-lowrider/25312/18

In any way, i will find a workaround if this challenge only concerns my CNC.

Thanks again.
